crates/proc/src/db.rs

```diff
@@ -912,13 +912,15 @@ fn disambiguate_add_year_suffix(tree: &mut IrTree, ctx: &CiteContext<'_, Markup>
         break;
     }
     if added_suffix {
+        tree.recompute_group_vars();
         return;
     }
 
     // Then attempt to do it for the ones that are embedded in date output
```

Comment:
This was the main missing piece. Explicit year suffixes weren't recomputing the tree's groupvars, only implicit ones were.